cli/src/core/parse/validate.go

```diff
@@ -0,0 +1,85 @@
+package parse
+
+import (
+	"path"
+	"strings"
+
+	"cli/core"
+	"cli/util/slice"
+
+	"github.com/luno/jettison/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	ErrNoConfigImage            = errors.New("config file missing field 'image'")
+	ErrNoPackages               = errors.New("no packages or custom packages specified")
+	ErrUndefinedProfilePackages = errors.New("packages in profile not in any of packages, custom-packages, or command-line custom-packages")
+	ErrNoSuchProfile            = errors.New("no such profile")
+)
+
+func validate(cmd *cobra.Command, config *core.Config) error {
+	customPackagePaths, err := cmd.Flags().GetStringSlice("custom-path")
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	if config.Image == "" {
+		return errors.Wrap(ErrNoConfigImage, "")
+	} else if len(config.Packages) == 0 && len(config.CustomPackages) == 0 && len(customPackagePaths) == 0 {
+		return errors.Wrap(ErrNoPackages, "")
+	}
+
+	return validateProfile(cmd, config)
+}
+
+func validateProfile(cmd *cobra.Command, config *core.Config) error {
+	profile, err := cmd.Flags().GetString("profile")
+	if err != nil {
+		return errors.Wrap(err, "")
+	}
+
+	profilePackagesMap := make(map[string]bool)
+	for _, prof := range config.Profiles {
+		if prof.Name == profile {
+			for _, p := range prof.Packages {
+				profilePackagesMap[p] = true
+			}
+		}
+	}
```

Comment:
Should we not have a break here?

Reply:
Done in [c1c77e1](https://github.com/openhie/package-starter-kit/pull/36/commits/c1c77e1d241de4a827525696efd5af5d945fb93e)